### PR TITLE
Announce media type for most watched items when using screen readers

### LINKED ISSUE #7630: Display Duration in the Most Watched Component Promo
**Is your feature request related to a problem? Please describe.**

__As a ws user, Maria,__
I want to know how long a piece of audio or video is 
so that I can decide whether I have the time to watch or listen to it 

__As a ws screenreader user, Carlos,__
I want to know whether a piece of content is a video or a piece of audio
so that I can make an informed decision as to whether I consume it 

---

**Describe the solution you'd like**

__Acceptance criteria:__
The most watched promo will display the clip duration where duration is present  
The most watched h3 label announces duration to screenreaders 
The most watched h3 label announces "video" where the clip is a video to screenreaders 
The most watched h3 label announces "listen" where the clip is audio to screenreaders

__Video Promo__

- Given I am using voiceover on IOS 
- when I navigate to the h3 link within the most watched promo list
- and the link is to a MAP with a video
- then the screenreader announces “video, 'headline', duration 'duration time', heading level 3, link” (e.g. “video, “News of the Day”, duration 05,00, heading level 3, link”) (semantic role order depends on the assistive tech)
- and "video", "duration" are announced in the language of the service
- and "heading level 3" and "link" are announced in the language recognised by the screenreader

__Audio Promo__

- Given I am using voiceover on IOS 
- when I navigate to the h3 link within the most watched promo list
- and the link is to a MAP with an audio clip 
- then the screenreader announces “Listen, 'headline', duration 'duration time', heading level 3, link” (e.g. “video, “News of the Day”, duration 05,00, heading level 3, link”) (semantic role order depends on the assistive tech)
- and "listen", "duration" are announced in the language of the service
- and "heading level 3" and "link" are announced in the language recognised by the screenreader

---

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by pharingee**
Resolves #7630

**Overall change:**
Announce media type for most watched items when using screen readers

**Code changes:**

-  The StoryPromo component announces the media type based on `cpsType`. Change the  condition for MAP to be more accurate by looking for `media` key.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/StoryPromo/LinkContents/index.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const LinkContents = ({ item, isInline }) => {
     translations: { media: mediaTranslations },
   } = useContext(ServiceContext);
 
-  const isMedia = pathOr(null, ['cpsType'], item) === 'MAP';
+  const isMedia = !!pathOr(null, ['media'], item);
```
Comment (karinathomasbbc) [inline src/app/containers/StoryPromo/LinkContents/index.jsx]: Is this a bit more readable?
```suggestion
  const isMedia = pathOr(false, ['media'], item);
```

**Review by karinathomasbbc — APPROVED** [review]

**Comment by rebeccamcginn** [comment]
"Change the condition for MAP to be more accurate." - Please can you explain what the problem is and how this solves it?

File: src/app/containers/StoryPromo/LinkContents/index.jsx (hunk 1 above)
Comment (rebeccamcginn) [reply src/app/containers/StoryPromo/LinkContents/index.jsx]: These are not the same; One is checking for the true/false association, the other returns the value at that path. You can use hasPath to return true/false.

**Comment by pharingee** [comment]
> "Change the condition for MAP to be more accurate." - Please can you explain what the problem is and how this solves it?

The problem is that the LinkContents component adds the extra code needed to announce a media type, but the condition for determining the media type depends only on the cpsType flag. This updates it to ensure that it also looks for a `media` object in promo as that is also a valid way of determining what a MAP is

**Review by rebeccamcginn — APPROVED** [review]

**Comment by LilyL0u** [comment]
Tested on Talkback - The media type is announced, e.g 'video' or 'audio', but the H3 link appears twice in the navigation sequence. It goes reads graphic -> reads H3 link -> reads H3 link -> reads date.

Did this come up in the a11y swarm or has this been introduced more recently?

**Pull request merged**